unittests/data/type_traits.hpp

```diff
@@ -544,13 +586,14 @@ namespace yes{
     struct x{
         x(){}
     };
+#ifdef __castxml__
+    typedef details::const_item const_item;
+    typedef details::const_container const_container;
+#endif
```

Comment:
Reading Brad's comment, this only works with ```c++11```. You could hijack the config to let the tests run with that flag if you need it (something like this: https://github.com/gccxml/pygccxml/blob/develop/unittests/test_cpp_standards.py#L32-L33). Maybe this part of the test could be extracted to run with the two flags, to test the two cases (broken vs non broken).